src/pool/cli.c

```diff
@@ -30,6 +30,9 @@ struct rsvc_client_state {
 	struct dc_mgmt_sys *scs_sys;
 };
 
+/** DAOS metrics pool rpc counters */
+daos_metrics_cntr_t	*pool_rpc_cntrs;
+
```

Comment:
these rpc_cnts  probably should be moved to src/client/api/metrics.c, then call these incr_xxx_cntr(module_id, opc) ?

Reply:
Please see my previous comment on why I moved completecntr update from daos_rpc_cb() to the higher level cb routines. Other reason to have these counters within the pool, cont and obj modules include the initialization (eg. dc_pool_metrics_init()) which can be better done from the respective module.

Reply:
Hmm, I mean each module (obj, pool, cont) can register its metrics process callback during dc_xxx_metrics_init(), then it can call dc_rpc_complete_metric_update(opc, ret) in each of rpc callback,  which will call these registered callback.  

1. put metrics related stuff in an individual file/functions as much as possible.
2. it would be easier to enable/disable metrics for certain module, for example if the user only want to measure the rpc for object module.

Reply:
Sure, let me think over it and do the rework.